src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionInfo.cs

```diff
@@ -39,17 +41,20 @@ public ConnectionInfo(ISqlConnectionFactory factory, string ownerUri, Connection
         /// <summary>
         /// Factory used for creating the SQL connection associated with the connection info.
         /// </summary>
-        public ISqlConnectionFactory Factory {get; private set;}
+        public ISqlConnectionFactory Factory { get; private set; }
 
         /// <summary>
         /// Properties used for creating/opening the SQL connection.
         /// </summary>
         public ConnectionDetails ConnectionDetails { get; private set; }
-        
+
         /// <summary>
-        /// The connection to the SQL database that commands will be run against.
+        /// A map containing all connections to the database that are associated with 
+        /// this ConnectionInfo's OwnerUri.
+        /// This is internal for testing access only
         /// </summary>
-        public DbConnection SqlConnection { get; set; }
+        internal readonly Dictionary<string, DbConnection> ConnectionTypeToConnectionMap =
```

Comment:
It was originally an enum and Kevin wanted it changed:

"We can't control the connection types. Instead use a string mapping, and use some constants file with Default and Query as the existing options. That way any 3rd party service can register their own connection type for a URI without changing the core code. You could potentially accept null and map to Default too"